### PR TITLE
[Chat] Update stateful client to return an empty string in failure cases

### PR DESCRIPTION
**Pull request by JoshuaLai**
# What
When downloading resources the stateful client will return an empty string
The selectors will now be responsible for reading the error object in the result and manipulate the html tags appropriatly

# Why
No more needing to default the stateful client to return a src url that is not correct. 

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/chat-component-bindings/src/messageThreadSelector.ts [hunk 1]
```diff
@@ -185,7 +190,12 @@ const processChatMessageContent = (message: ChatMessageWithStatus): string | und
 const generateImageAttachmentImgHtml = (message: ChatMessageWithStatus, attachment: ChatAttachment): string => {
   if (attachment.previewUrl !== undefined) {
     const contentType = extractAttachmentContentTypeFromName(attachment.name);
-    const src = message.resourceCache?.[attachment.previewUrl].sourceUrl ?? '';
+    const resourceCache = message.resourceCache?.[attachment.previewUrl];
+    let src = URL.createObjectURL(new Blob());
```
Comment (emlynmac) [inline packages/chat-component-bindings/src/messageThreadSelector.ts:194]: Why do we need to create a new empty blob here?
How do you free this up later?

Comment (emlynmac) [reply packages/chat-component-bindings/src/messageThreadSelector.ts:194]: Why can this not just be an empty string?

Comment (JoshuaLai) [reply packages/chat-component-bindings/src/messageThreadSelector.ts]: Resolved with using just a static string of `blob://` for now

**Review by emlynmac — APPROVED** [review]

File: packages/chat-component-bindings/src/messageThreadSelector.ts [hunk 2]
```diff
@@ -185,7 +190,12 @@ const processChatMessageContent = (message: ChatMessageWithStatus): string | und
 const generateImageAttachmentImgHtml = (message: ChatMessageWithStatus, attachment: ChatAttachment): string => {
   if (attachment.previewUrl !== undefined) {
     const contentType = extractAttachmentContentTypeFromName(attachment.name);
-    const src = message.resourceCache?.[attachment.previewUrl].sourceUrl ?? '';
+    const resourceCache = message.resourceCache?.[attachment.previewUrl];
+    let src = 'blob://';
+    if (resourceCache && resourceCache.error === undefined) {
+      src = resourceCache.sourceUrl;
+    }
```
Comment (palatter) [inline packages/chat-component-bindings/src/messageThreadSelector.ts:197]: This block is repeated, could we move it to it's own function?

**Review by palatter — APPROVED** [review]
Looks good, one small comment

**Pull request merged**